### ISSUE TITLE
teacher cant copy glossary

### ISSUE THREAD

**Issue by Mikkeljeppesen1** [praxisdigital/moodle-block_sharing_cart#101]
Found on rybners:

If youre a teacher and try to copy a glossary into the sharingcart you get a pop that says something like: do you want to copy userdata. If i mark it, the userdata still wont getting copied. Its not a capabilityissue but rather allowing copy the data. If i create a manual backup of the glossary as an xml file i can.

**Comment by mcardle** [comment]
I think we have had this issue before. The glossary is saved as user data (as far as I know), but I think I remember something about it was not possible to support in the sharing cart, but you can create an export from the original course and import it manually. Try searching for the issue in Hubspot and see if you are able to find it.

**Comment by AraPraxis** [comment]
@Mikkeljeppesen1 
I've logged into Rybners and tested aswell.

- The admin user has the capability to copy glossary activities. When copied to a course the user data is saved aswell.

- Rybners' teacher role has the capability to copy glossary activities, but I personally do not see the checkbox as Mikkel mentions..?


I've looked into the capabilities and found out that the following capabilities are missing on the teacher role (I've chosen to test but not set the capability permanently, as I do not know if the customer wants this..);


![Image](https://github.com/praxisdigital/moodle-block_sharing_cart/assets/89771146/d745d163-2eff-4cd6-9b39-b3108d1d57d8)


After turning them on for a short period of time, I could copy and save the glossary activity, with the user data, as a teacher.
I personally do not see a bug on their site, but might need to see how the bug occurs, if the information above does not help.
